src/server/v2.0/handler/scanexport.go

```diff
@@ -0,0 +1,202 @@
+package handler
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/strfmt"
+	"github.com/goharbor/harbor/src/common/rbac"
+	"github.com/goharbor/harbor/src/controller/scandataexport"
+	"github.com/goharbor/harbor/src/jobservice/logger"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/lib/orm"
+	"github.com/goharbor/harbor/src/pkg/scan/export"
+	v1 "github.com/goharbor/harbor/src/pkg/scan/rest/v1"
+	"github.com/goharbor/harbor/src/pkg/systemartifact"
+	"github.com/goharbor/harbor/src/server/v2.0/models"
+	operation "github.com/goharbor/harbor/src/server/v2.0/restapi/operations/scan_data_export"
+)
+
+func newScanDataExportApi() *scanDataExportAPI {
+	return &scanDataExportAPI{
+		scanDataExportCtl: scandataexport.Ctl,
+		sysArtifactMgr:    systemartifact.Mgr,
+	}
+}
+
+type scanDataExportAPI struct {
+	BaseAPI
+	scanDataExportCtl scandataexport.Controller
+	sysArtifactMgr    systemartifact.Manager
+}
+
+func (se *scanDataExportAPI) Prepare(ctx context.Context, operation string, params interface{}) middleware.Responder {
+	return nil
+}
+
+func (se *scanDataExportAPI) ExportScanData(ctx context.Context, params operation.ExportScanDataParams) middleware.Responder {
+	if err := se.RequireAuthenticated(ctx); err != nil {
+		return se.SendError(ctx, err)
+	}
+
+	// check if the MIME type for the export is the Generic vulnerability data
+	if params.XScanDataType != v1.MimeTypeGenericVulnerabilityReport {
+		error := &models.Error{Message: fmt.Sprintf("Unsupported MIME type : %s", params.XScanDataType)}
+		errors := &models.Errors{Errors: []*models.Error{error}}
+		return operation.NewExportScanDataBadRequest().WithPayload(errors)
+	}
+
+	// loop through the list of projects and validate that scan privilege and create privilege
+	// is available for all projects
+	// TODO : Should we just ignore projects that do not have the required level of access?
+
+	projects := params.Criteria.Projects
+	for _, project := range projects {
+		if err := se.RequireProjectAccess(ctx, project, rbac.ActionCreate, rbac.ResourceScan); err != nil {
```

Comment:
is it a system level API? If so, only system access needs to be verified.

Reply:
As discussed, there is no change required here. Skipping this comment.